packages/core/middlewares.ts

```diff
@@ -19,26 +21,53 @@ export namespace Middlewares {
     return parser(req, res, next);
   }
 
-  export function Preflight(req, res, next) {
-    res.header("Access-Control-Allow-Origin", req.header("Origin") || "*");
+  export function Preflight(options: CorsOptions) {
+    return (req, res, next) => {
+      let allowedOrigin = getMatchedValue(req.header("Origin"), options.allowedOrigins);
 
-    if (req.header("access-control-request-method")) {
-      res.header("Access-Control-Allow-Methods", req.header("access-control-request-method"));
-    }
+      res.header("Access-Control-Allow-Origin", allowedOrigin);
 
-    let allowedHeaders = "Authorization, Content-Type, Accept-Language";
-    if (req.header("access-control-request-headers")) {
-      allowedHeaders = req.header("access-control-request-headers");
-    }
+      //console.log(res.header("Access-Control-Allow-Origin"));
 
-    res.header("Access-Control-Allow-Headers", allowedHeaders);
+      let allowedMethod = getMatchedValue(
+        req.header("access-control-request-method")
+          ? req.header("access-control-request-method")
+          : req.method,
+        options.allowedMethods
+      );
 
-    req.header("Access-Control-Allow-Credentials", "true");
+      res.header("Access-Control-Allow-Methods", allowedMethod);
 
-    if (req.method == "OPTIONS") {
-      res.end();
-    } else {
-      next();
-    }
+      let allowedHeaders = options.allowedHeaders.join(",");
+
+      if (req.header("access-control-request-headers")) {
+        allowedHeaders = getMatchedValue(
+          req.header("access-control-request-headers").indexOf(",") != -1
+            ? req.header("access-control-request-headers").split(",")
+            : req.header("access-control-request-headers"),
```

Comment:
```suggestion
          req.header("access-control-request-headers").split(",")
```
Could we just use the array all the time? I mean if there is no comma char in the string this logic will return an array whose first item's content is the content of the header. So I think may not a logic like that